src/main/scala/is/hail/asm4s/FunctionBuilder.scala

```diff
@@ -28,21 +34,29 @@ abstract class FunctionBuilder[R](parameterTypeInfo: Array[TypeInfo[_]], returnT
   cn.version = V1_8
   cn.access = ACC_PUBLIC
 
-  cn.name = packageName + "/C" + newUniqueID()
+  val name = packageName + "/C" + newUniqueID()
+  cn.name = name
   cn.superName = "java/lang/Object"
+  cn.interfaces.asInstanceOf[java.util.List[String]].add("java/io/Serializable")
 
-  def signature: String = s"(${ parameterTypeInfo.map(_.name).mkString })${ returnTypeInfo.name }"
+  def descriptor: String = s"(${ parameterTypeInfo.map(_.name).mkString })${ returnTypeInfo.name }"
 
-  val mn = new MethodNode(ACC_PUBLIC + ACC_STATIC, "f", signature, null, null)
+  val mn = new MethodNode(ACC_PUBLIC, "apply", descriptor, null, null)
+  val init = new MethodNode(ACC_PUBLIC, "<init>", "()V", null, null)
   // FIXME why is cast necessary?
   cn.methods.asInstanceOf[util.List[MethodNode]].add(mn)
+  cn.methods.asInstanceOf[util.List[MethodNode]].add(init)
   val il = mn.instructions
```

Comment:
The `(...).add(init)`?

This is implicitly called by the constructor when we call `loadClass(...).newInstance()`. If the `init` method doesn't exist the JVM errors out.